rclcpp/test/rclcpp/test_qos_event.cpp

```diff
@@ -207,7 +209,9 @@ TEST_F(TestQosEvent, test_default_incompatible_qos_callbacks)
   rclcpp::executors::SingleThreadedExecutor ex;
   ex.add_node(node->get_node_base_interface());
 
-  ex.spin_until_future_complete(log_msgs_future, std::chrono::seconds(10));
+  // This future won't complete on fastrtps, so just timeout immediately
+  const auto timeout = (is_fastrtps) ? std::chrono::milliseconds(5) : std::chrono::seconds(10);
+  ex.spin_until_future_complete(log_msgs_future, timeout);
```

Comment:
Thinking about this test more, can't we just skip it completely?  That is, at the top of this method, just have:

```
if (is_fastrtps) {
  return;
}
```

Reply:
gtest does have a macro for runtime skipping of tests (`GTEST_SKIP`), but that would make it harder for this issue to be detected in the future if the code changes for fastrtps.

I updated the test to just check that the logger messages are empty.